internal/ybdata/ybdata.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 YourBase Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// Package ybdata locates directories the user has designated or conventionally
+// uses for storing different types of data.
+package ybdata
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"go4.org/xdgdir"
+)
+
+// Dirs is the result of locating directories.
+type Dirs struct {
+	cache      string
+	workspaces string
+}
+
+// FromEnv finds data directories based on environment variables.
+func FromEnv() (*Dirs, error) {
```

Comment:
Based on Effective Go's [_Package Names_][1] section the name `ybdata.FromEnv` makes me believe `FromEnv` should return "a ybdata". I think this could flow better if the function was called `DirsFromEnv`, or if the package was renamed to `dirs` or `ybdirs` so you could call `dirs.New` and `dirs.FromEnv`.

[1]: https://golang.org/doc/effective_go.html#package-names